### DISCUSSION TITLE
doris-flink-connector能提供编译好的jar包吗，编译失败

### DISCUSSION THREAD

**Discussion by ztyd** [apache/doris#8430] [A - General / Q&A]
始终编译失败：
[ERROR] error: scala.reflect.internal.MissingRequirementError: object java.lang.Object in compiler mirror not found.
[ERROR]         at scala.reflect.internal.MissingRequirementError$.signal(MissingRequirementError.scala:17)
[ERROR]         at scala.reflect.internal.MissingRequirementError$.notFound(MissingRequirementError.scala:18)
[ERROR]         at scala.reflect.internal.Mirrors$RootsBase.$anonfun$getModuleOrClass$4(Mirrors.scala:54)
[ERROR]         at scala.reflect.internal.Mirrors$RootsBase.getModuleOrClass(Mirrors.scala:54)
[ERROR]         at scala.reflect.internal.Mirrors$RootsBase.getModuleOrClass(Mirrors.scala:45)
[ERROR]         at scala.reflect.internal.Mirrors$RootsBase.getModuleOrClass(Mirrors.scala:45)
[ERROR]         at scala.reflect.internal.Mirrors$RootsBase.getModuleOrClass(Mirrors.scala:66)
[ERROR]         at scala.reflect.internal.Mirrors$RootsBase.getClassByName(Mirrors.scala:102)
[ERROR]         at scala.reflect.internal.Mirrors$RootsBase.getRequiredClass(Mirrors.scala:105)
[ERROR]         at scala.reflect.internal.Definitions$DefinitionsClass.ObjectClass$lzycompute(Definitions.scala:267)
[ERROR]         at scala.reflect.internal.Definitions$DefinitionsClass.ObjectClass(Definitions.scala:267)
[ERROR]         at scala.reflect.internal.Definitions$DefinitionsClass.init(Definitions.scala:1447)
[ERROR]         at scala.tools.nsc.Global$Run.<init>(Global.scala:1126)
[ERROR]         at scala.tools.nsc.MainClass.doCompile(Main.scala:24)
[ERROR]         at scala.tools.nsc.Driver.process(Driver.scala:55)
[ERROR]         at scala.tools.nsc.Driver.main(Driver.scala:68)
[ERROR]         at scala.tools.nsc.Main.main(Main.scala)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR]         at scala_maven_executions.MainHelper.runMain(MainHelper.java:164)
[ERROR]         at scala_maven_executions.MainWithArgsInFile.main(MainWithArgsInFile.java:26)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------

[ERROR] Failed to execute goal net.alchim31.maven:scala-maven-plugin:3.2.1:compile (scala-compile-first) on project doris-flink: wrap: org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1) -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
各种版本都试过了
能提供编译好的jar包吗，官网的maven依赖
<dependency>
  <groupId>org.apache.doris</groupId>
  <artifactId>doris-flink-connector</artifactId>
  <version>1.11.6-2.12-SNAPSHOT</version>
</dependency>
下不到文件

**Comment by hf200012** [comment]
It will be pushed to the maven central repository soon

**Reply by ztyd** [reply]
good news

**Reply by hf200012** [reply]
```
<dependency>
  <groupId>org.apache.doris</groupId>
  <artifactId>flink-doris-connector-1.14_2.12</artifactId>
  <!--artifactId>flink-doris-connector-1.13_2.12</artifactId-->
  <!--artifactId>flink-doris-connector-1.12_2.12</artifactId-->
  <!--artifactId>flink-doris-connector-1.11_2.12</artifactId-->
  <version>1.0.3</version>
</dependency>
```

**Reply by huyuanfeng2018** [reply]
> 

Can these jar packages support doris 1.1.1?

**Reply by JNSimba** [reply]
![image](https://user-images.githubusercontent.com/7951521/186404484-c09e7be5-5ce8-4080-9b0e-c7da8995c0a2.png)

**Reply by huyuanfeng2018** [reply]
Our current flink version is 1.12, but we are currently using doris 1.1.1. Our current needs are only writing. I don’t know if it can be supported

**Reply by huyuanfeng2018** [reply]
> Our current flink version is 1.12, but we are currently using doris 1.1.1. Our current needs are only writing. I don’t know if it can be supported

i need 2pc

**Comment by fwchitu** [comment]
你可以用doris官网的jar包试一试，有问题，回复到这里就行。1.1.1应该是支持的。

**Reply by huyuanfeng2018** [reply]
其实我们现在大批量用的是1.12版本的flink，升级比较麻烦，但是我们想要数据一致性，但我看connector发的release里面connector1.1以后没有对于flink1.12版的支持，我用master分支指定flink1.12无法编译通过，有些依赖不兼容。所以想问下对flink老版本能不能也支持一下最新的这个eos特性

**Reply by huyuanfeng2018** [reply]
我没细看这个connector的代码，不知道为啥没有支持到1.12版本。我的理解是理论上是可以支持到flink1.12版本的，如果说不是特别麻烦的话，我愿意贡献并且维护这个1.12版本的分支。

**Comment by fwchitu** [comment]
嗯，好呀，这些代码是开源的，到时让相关负责这块的东西联系你哈，欢迎为社区做出贡献哈。

**Comment by JNSimba** [comment]
Hello，Can you add my WeChat (jnsimba), let's take a look at the problem of flink1.12 supporting eos.
